src/main/scala/it/pagopa/interop/backendforfrontend/error/Handlers.scala

```diff
@@ -1,92 +1,89 @@
 package it.pagopa.interop.backendforfrontend.error
 
-import akka.http.scaladsl.marshalling.ToEntityMarshaller
 import akka.http.scaladsl.model.StatusCodes
-import akka.http.scaladsl.server.Directives.complete
 import akka.http.scaladsl.server.StandardRoute
 import com.typesafe.scalalogging.LoggerTakingImplicit
 import it.pagopa.interop.agreementprocess.client.invoker.{ApiError => AgreementProcessError}
 import it.pagopa.interop.attributeregistrymanagement.client.invoker.{ApiError => AttributeRegistryError}
-import it.pagopa.interop.backendforfrontend.api.impl.{problemFormat, problemOf}
-import it.pagopa.interop.backendforfrontend.error.BFFErrors.AttributeNotExists
+import it.pagopa.interop.backendforfrontend.api.impl.{problemFormat, problemOf, serviceCode}
+import it.pagopa.interop.backendforfrontend.error.BFFErrors.{AttributeNotExists, DownstreamError}
 import it.pagopa.interop.backendforfrontend.model.Problem
 import it.pagopa.interop.catalogmanagement.client.invoker.{ApiError => CatalogManagementError}
 import it.pagopa.interop.catalogprocess.client.invoker.{ApiError => CatalogProcessError}
 import it.pagopa.interop.commons.logging.ContextFieldsToLog
 import it.pagopa.interop.commons.ratelimiter
-import it.pagopa.interop.commons.ratelimiter.model.{Headers, RateLimitStatus}
-import it.pagopa.interop.commons.utils.errors.GenericComponentErrors.{GenericError, TooManyRequests}
+import it.pagopa.interop.commons.ratelimiter.model.Headers
+import it.pagopa.interop.commons.utils.errors.AkkaResponses._
+import it.pagopa.interop.commons.utils.errors.GenericComponentErrors.GenericError
+import it.pagopa.interop.commons.utils.errors.{ComponentError, GenericComponentErrors, ServiceCode}
 import it.pagopa.interop.selfcare.partyprocess.client.invoker.{ApiError => PartyProcessError}
 import it.pagopa.interop.selfcare.userregistry.client.invoker.{ApiError => UserRegistryError}
 import it.pagopa.interop.tenantmanagement.client.invoker.{ApiError => TenantManagementError}
 import it.pagopa.interop.tenantprocess.client.invoker.{ApiError => TenantProcessError}
 import spray.json._
 
+import scala.util.matching.Regex
 import scala.util.{Failure, Try}
 
 object Handlers {
 
+  val downstreamErrorCodeRegex: Regex = "^([0-9]+)-([0-9]+)$".r
```

Comment:
Regexes are awesome until they aren't. It's mandatory to ask: don't we have another way to extract the 2 codes? It's a fairly simple regex but you know... it's a regex :D

Reply:
fair enough! I will split by `-` and pattern match on the resulting list 👍